frontend/dockerfile/dockerfile_test.go

```diff
@@ -6310,6 +6311,61 @@ COPY --from=build out .
 	require.Equal(t, "freebsd", string(dt))
 }
 
+func testPlatformArgsExplicitContext(t *testing.T, sb integration.Sandbox) {
+	integration.SkipOnPlatform(t, "windows")
+	f := getFrontend(t, sb)
+
+	dockerfile := []byte(`
+FROM --platform=$BUILDPLATFORM scratch AS build
+COPY --from=bb . /
+ARG TARGETPLATFORM
+ARG TARGETOS
+RUN mkdir /out && echo -n $TARGETPLATFORM > /out/platform && echo -n $TARGETOS > /out/os
+FROM scratch
+COPY --from=build out .
+`)
+
+	dir := integration.Tmpdir(
+		t,
+		fstest.CreateFile("Dockerfile", dockerfile, 0600),
+	)
+
+	c, err := client.New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	destDir := t.TempDir()
+
+	opt := client.SolveOpt{
+		Exports: []client.ExportEntry{
+			{
+				Type:      client.ExporterLocal,
+				OutputDir: destDir,
+			},
+		},
+		FrontendAttrs: map[string]string{
+			"platform":           "darwin/ppc64le",
+			"build-arg:TARGETOS": "freebsd",
+			"context:bb":         "docker-image://busybox",
```

Comment:
With this config you would need a busybox image for `darwin/ppc64le`.

Dockerfile should say.

```
FROM --platform=$BUILDPLATFORM bb AS other


....

COPY --from=other
```